### DISCUSSION TITLE
NaN Jaccard When Negative Sampling

### DISCUSSION THREAD

**Discussion by vjsrinivas** [Lightning-AI/torchmetrics#1414] [CompVision]
Hello,

I've been trying to use the JaccardIndex on my binary segmentation task. It's been working as expected with data that contains the objects I'm segmenting. To improve robustness, I add negative samples to the mix. Since its negative sampling, I produce a prediction mask with all zeros and a ground truth with all zeros. When this happens, JaccardIndex outputs NaN (assuming its caused by 0/0).

```python
pred = torch.zeros((1,1,100,100))
true = torch.zeros((1,1,100,100))
jaccard = JaccardIndex(num_classes=1, task="binary")
iou = jaccard(_pred, _true)
print(iou)
```

Other metric packages seem to deal with it by outputting zero: https://github.com/drostlab/philentropy/issues/15#issuecomment-457306889
Is there a way to mimic this behavior?

Apologies if there was a similar topic. I searched for one and couldn't find any.
